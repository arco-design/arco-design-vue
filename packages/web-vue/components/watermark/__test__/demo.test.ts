import demoTest from '../../../scripts/demo-test';

demoTest('watermark');
