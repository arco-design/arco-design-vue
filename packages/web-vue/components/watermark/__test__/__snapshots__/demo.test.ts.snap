// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<watermark> demo: render [basic] correctly 1`] = `
"<div class=\\"arco-watermark\\" style=\\"position: relative; overflow: hidden;\\">
  <div style=\\"width: 100%; height: 350px;\\"></div>
</div>"
`;

exports[`<watermark> demo: render [custom] correctly 1`] = `
"<form class=\\"arco-form arco-form-layout-horizontal arco-form-size-small arco-form-auto-label-width\\">
  <div class=\\"arco-row arco-row-align-start arco-row-justify-start\\" style=\\"margin-left: -8px; margin-right: -8px;\\">
    <div class=\\"arco-col arco-col-24\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
      <div class=\\"arco-row arco-row-nowrap arco-row-align-start arco-row-justify-start arco-form-item arco-form-item-layout-horizontal\\">
        <div class=\\"arco-col arco-col-5 arco-form-item-label-col arco-form-item-label-col-flex\\"><label class=\\"arco-form-item-label\\">
            <!--v-if-->rotate
            <!--v-if-->
            <!--v-if-->
          </label></div>
        <div class=\\"arco-col arco-form-item-wrapper-col\\" style=\\"flex-basis: auto;\\" id=\\"rotate\\">
          <div class=\\"arco-form-item-content-wrapper\\">
            <div class=\\"arco-form-item-content arco-form-item-content-flex\\">
              <div class=\\"arco-slider\\">
                <div class=\\"arco-slider-track\\">
                  <div class=\\"arco-slider-bar\\" style=\\"left: 0%; right: 50%;\\"></div>
                  <!--v-if-->
                  <!--v-if-->
                  <!--v-if-->
                  <!--v-if-->
                  <div style=\\"left: 50%;\\" tabindex=\\"0\\" role=\\"slider\\" aria-disabled=\\"false\\" aria-valuemax=\\"180\\" aria-valuemin=\\"-180\\" aria-valuenow=\\"0\\" aria-valuetext=\\"0\\" class=\\"arco-slider-btn\\"></div>
                  <!---->
                </div>
                <!--v-if-->
              </div>
            </div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class=\\"arco-col arco-col-12\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
      <div class=\\"arco-row arco-row-nowrap arco-row-align-start arco-row-justify-start arco-form-item arco-form-item-layout-horizontal\\">
        <div class=\\"arco-col arco-col-5 arco-form-item-label-col arco-form-item-label-col-flex\\"><label class=\\"arco-form-item-label\\">
            <!--v-if-->gap
            <!--v-if-->
            <!--v-if-->
          </label></div>
        <div class=\\"arco-col arco-form-item-wrapper-col\\" style=\\"flex-basis: auto;\\">
          <div class=\\"arco-form-item-content-wrapper\\">
            <div class=\\"arco-form-item-content arco-form-item-content-flex\\">
              <div class=\\"arco-input-group\\"><span class=\\"arco-input-wrapper arco-input-number arco-input-number-mode-embed arco-input-number-size-small\\"><!----><input class=\\"arco-input arco-input-size-small\\" type=\\"text\\" placeholder=\\"gap[x]\\" role=\\"spinbutton\\" aria-valuemax=\\"Infinity\\" aria-valuemin=\\"0\\" aria-valuenow=\\"50\\"><!----><span class=\\"arco-input-suffix\\"><!----><!----><div class=\\"arco-input-number-step\\"><button class=\\"arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-up\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 30.557 24.043 15 8.487 30.557\\"></path></svg></button><button class=\\"arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-down\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></button></div><!----></span></span><span class=\\"arco-input-wrapper arco-input-number arco-input-number-mode-embed arco-input-number-size-small\\"><!----><input class=\\"arco-input arco-input-size-small\\" type=\\"text\\" placeholder=\\"gap[y]\\" role=\\"spinbutton\\" aria-valuemax=\\"Infinity\\" aria-valuemin=\\"0\\" aria-valuenow=\\"50\\"><!----><span class=\\"arco-input-suffix\\"><!----><!----><div class=\\"arco-input-number-step\\"><button class=\\"arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-up\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 30.557 24.043 15 8.487 30.557\\"></path></svg></button><button class=\\"arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-down\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></button></div><!----></span></span></div>
            </div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class=\\"arco-col arco-col-12\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
      <div class=\\"arco-row arco-row-nowrap arco-row-align-start arco-row-justify-start arco-form-item arco-form-item-layout-horizontal\\">
        <div class=\\"arco-col arco-col-5 arco-form-item-label-col arco-form-item-label-col-flex\\"><label class=\\"arco-form-item-label\\">
            <!--v-if-->offset
            <!--v-if-->
            <!--v-if-->
          </label></div>
        <div class=\\"arco-col arco-form-item-wrapper-col\\" style=\\"flex-basis: auto;\\">
          <div class=\\"arco-form-item-content-wrapper\\">
            <div class=\\"arco-form-item-content arco-form-item-content-flex\\">
              <div class=\\"arco-input-group\\"><span class=\\"arco-input-wrapper arco-input-number arco-input-number-mode-embed arco-input-number-size-small\\"><!----><input class=\\"arco-input arco-input-size-small\\" type=\\"text\\" placeholder=\\"offsetLeft\\" role=\\"spinbutton\\" aria-valuemax=\\"Infinity\\" aria-valuemin=\\"-Infinity\\" aria-valuenow=\\"\\"><!----><span class=\\"arco-input-suffix\\"><!----><!----><div class=\\"arco-input-number-step\\"><button class=\\"arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-up\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 30.557 24.043 15 8.487 30.557\\"></path></svg></button><button class=\\"arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-down\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></button></div><!----></span></span><span class=\\"arco-input-wrapper arco-input-number arco-input-number-mode-embed arco-input-number-size-small\\"><!----><input class=\\"arco-input arco-input-size-small\\" type=\\"text\\" placeholder=\\"offsetTop\\" role=\\"spinbutton\\" aria-valuemax=\\"Infinity\\" aria-valuemin=\\"-Infinity\\" aria-valuenow=\\"\\"><!----><span class=\\"arco-input-suffix\\"><!----><!----><div class=\\"arco-input-number-step\\"><button class=\\"arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-up\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 30.557 24.043 15 8.487 30.557\\"></path></svg></button><button class=\\"arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-down\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M39.6 17.443 24.043 33 8.487 17.443\\"></path></svg></button></div><!----></span></span></div>
            </div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class=\\"arco-col arco-col-12\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
      <div class=\\"arco-row arco-row-nowrap arco-row-align-start arco-row-justify-start arco-form-item arco-form-item-layout-horizontal\\">
        <div class=\\"arco-col arco-col-5 arco-form-item-label-col arco-form-item-label-col-flex\\"><label class=\\"arco-form-item-label\\">
            <!--v-if-->fontSize
            <!--v-if-->
            <!--v-if-->
          </label></div>
        <div class=\\"arco-col arco-form-item-wrapper-col\\" style=\\"flex-basis: auto;\\">
          <div class=\\"arco-form-item-content-wrapper\\">
            <div class=\\"arco-form-item-content arco-form-item-content-flex\\"><span class=\\"arco-input-outer arco-input-outer-size-small arco-input-number arco-input-number-mode-button arco-input-number-size-small\\"><span class=\\"arco-input-prepend\\"><button class=\\"arco-btn arco-btn-secondary arco-btn-shape-square arco-btn-size-small arco-btn-status-normal arco-btn-only-icon arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><span class=\\"arco-btn-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-minus\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M5 24h38\\"></path></svg></span></button></span><span class=\\"arco-input-wrapper\\"><!----><input class=\\"arco-input arco-input-size-small\\" type=\\"text\\" role=\\"spinbutton\\" aria-valuemax=\\"Infinity\\" aria-valuemin=\\"-Infinity\\" aria-valuenow=\\"16\\"><!----><!----></span><span class=\\"arco-input-append\\"><button class=\\"arco-btn arco-btn-secondary arco-btn-shape-square arco-btn-size-small arco-btn-status-normal arco-btn-only-icon arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><span class=\\"arco-btn-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-plus\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M5 24h38M24 5v38\\"></path></svg></span></button></span></span></div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class=\\"arco-col arco-col-12\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
      <div class=\\"arco-row arco-row-nowrap arco-row-align-start arco-row-justify-start arco-form-item arco-form-item-layout-horizontal\\">
        <div class=\\"arco-col arco-col-5 arco-form-item-label-col arco-form-item-label-col-flex\\"><label class=\\"arco-form-item-label\\">
            <!--v-if-->zIndex
            <!--v-if-->
            <!--v-if-->
          </label></div>
        <div class=\\"arco-col arco-form-item-wrapper-col\\" style=\\"flex-basis: auto;\\">
          <div class=\\"arco-form-item-content-wrapper\\">
            <div class=\\"arco-form-item-content arco-form-item-content-flex\\"><span class=\\"arco-input-outer arco-input-outer-size-small arco-input-number arco-input-number-mode-button arco-input-number-size-small\\"><span class=\\"arco-input-prepend\\"><button class=\\"arco-btn arco-btn-secondary arco-btn-shape-square arco-btn-size-small arco-btn-status-normal arco-btn-only-icon arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><span class=\\"arco-btn-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-minus\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M5 24h38\\"></path></svg></span></button></span><span class=\\"arco-input-wrapper\\"><!----><input class=\\"arco-input arco-input-size-small\\" type=\\"text\\" role=\\"spinbutton\\" aria-valuemax=\\"Infinity\\" aria-valuemin=\\"-Infinity\\" aria-valuenow=\\"6\\"><!----><!----></span><span class=\\"arco-input-append\\"><button class=\\"arco-btn arco-btn-secondary arco-btn-shape-square arco-btn-size-small arco-btn-status-normal arco-btn-only-icon arco-input-number-step-button\\" type=\\"button\\" tabindex=\\"-1\\"><span class=\\"arco-btn-icon\\"><svg viewBox=\\"0 0 48 48\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" stroke=\\"currentColor\\" class=\\"arco-icon arco-icon-plus\\" stroke-width=\\"4\\" stroke-linecap=\\"butt\\" stroke-linejoin=\\"miter\\"><path d=\\"M5 24h38M24 5v38\\"></path></svg></span></button></span></span></div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class=\\"arco-col arco-col-6\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
      <div class=\\"arco-row arco-row-nowrap arco-row-align-start arco-row-justify-start arco-form-item arco-form-item-layout-horizontal\\">
        <div class=\\"arco-col arco-col-5 arco-form-item-label-col arco-form-item-label-col-flex\\"><label class=\\"arco-form-item-label\\">
            <!--v-if-->repeat
            <!--v-if-->
            <!--v-if-->
          </label></div>
        <div class=\\"arco-col arco-form-item-wrapper-col\\" style=\\"flex-basis: auto;\\">
          <div class=\\"arco-form-item-content-wrapper\\">
            <div class=\\"arco-form-item-content arco-form-item-content-flex\\"><button type=\\"button\\" role=\\"switch\\" aria-checked=\\"true\\" class=\\"arco-switch arco-switch-type-circle arco-switch-checked\\"><span class=\\"arco-switch-handle\\"><span class=\\"arco-switch-handle-icon\\"></span></span><!--  prettier-ignore  -->
                <!--v-if-->
              </button></div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <div class=\\"arco-col arco-col-6\\" style=\\"padding-left: 8px; padding-right: 8px;\\">
      <div class=\\"arco-row arco-row-nowrap arco-row-align-start arco-row-justify-start arco-form-item arco-form-item-layout-horizontal\\">
        <div class=\\"arco-col arco-col-5 arco-form-item-label-col arco-form-item-label-col-flex\\"><label class=\\"arco-form-item-label\\">
            <!--v-if-->staggered
            <!--v-if-->
            <!--v-if-->
          </label></div>
        <div class=\\"arco-col arco-form-item-wrapper-col\\" style=\\"flex-basis: auto;\\">
          <div class=\\"arco-form-item-content-wrapper\\">
            <div class=\\"arco-form-item-content arco-form-item-content-flex\\"><button type=\\"button\\" role=\\"switch\\" aria-checked=\\"true\\" class=\\"arco-switch arco-switch-type-circle arco-switch-checked\\"><span class=\\"arco-switch-handle\\"><span class=\\"arco-switch-handle-icon\\"></span></span><!--  prettier-ignore  -->
                <!--v-if-->
              </button></div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
  </div>
</form>
<div class=\\"arco-watermark\\" style=\\"position: relative; overflow: hidden;\\">
  <div style=\\"width: 100%; border: 1px solid #e5e6eb; box-sizing: border-box;\\">
    <h5 class=\\"arco-typography\\"> Design system
      <!---->
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </h5>
    <article class=\\"arco-typography\\">
      <div class=\\"arco-typography\\"> A design is a plan or specification for the construction of an object or system or for the implementation of an activity or process, or the result of that plan or specification in the form of a prototype, product or process. The verb to design expresses the process of developing a design.
        <!---->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
      <div class=\\"arco-typography\\"> A design is a plan or specification for the construction of an object or system or for the implementation of an activity or process, or the result of that plan or specification in the form of a prototype, product or process. The verb to design expresses the process of developing a design.
        <!---->
        <!--v-if-->
        <!--v-if-->
        <!--v-if-->
      </div>
    </article><img style=\\"position: relative; z-index: 7;\\" src=\\"https://p1-arco.byteimg.com/tos-cn-i-uwbnlip3yd/24e0dd27418d2291b65db1b21aa62254.png~tplv-uwbnlip3yd-webp.webp\\">
  </div>
</div>"
`;

exports[`<watermark> demo: render [image] correctly 1`] = `
"<div class=\\"arco-watermark\\" style=\\"position: relative; overflow: hidden;\\">
  <div style=\\"width: 100%; height: 300px;\\"></div>
</div>"
`;

exports[`<watermark> demo: render [multiline] correctly 1`] = `
"<div class=\\"arco-watermark\\" style=\\"position: relative; overflow: hidden;\\">
  <div style=\\"width: 100%; height: 300px;\\"></div>
</div>"
`;
